src/lib/reasoners/satml.ml

```diff
@@ -460,7 +460,15 @@ module Make (Th : Theory.S) : SAT_ML with type th = Th.t = struct
     a.var.vpremise <- []
 
   let enqueue_assigned env a =
-    if a.neg.is_guard then raise (Unsat None);
+    if a.neg.is_guard then begin
+      (* if the negation of a is (still) a guard, it should be forced to true
+         during the first decisions.
+         If the SAT tries to deduce that a.neg is true (ie. a is false),
+         then we have detected an inconsistency. *)
+      assert (a.var.level <= env.next_dec_guard);
```

Comment:
Is it `env.next_dec_guard` or just `Vec.size env.increm_guards` ?

Reply:
I think it's better to use `env.next_dec_guard`. This value is incremented when we decide (`pick_branch_lit`) a guard. We want to ensure that if we enqueue `a` and `a.neg` is a guard, this guard `a.neg` was decided before the current next_guard_lvl.